From b6750350161b661cd3dd6fba3ce5fa88897bfd6d Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Wed, 16 Apr 2025 17:28:08 +0900
Subject: [PATCH 6/9] HACK: Skip fdt_shrink_to_minimum to avoid stack

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 common/spl/spl.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/common/spl/spl.c b/common/spl/spl.c
index 76fd56dfe4b..50d11dc7d9b 100644
--- a/common/spl/spl.c
+++ b/common/spl/spl.c
@@ -140,11 +140,12 @@ void spl_fixup_fdt(void *fdt_blob)
 	}
 
 	/* fixup the memory dt node */
-	err = fdt_shrink_to_minimum(fdt_blob, 0);
-	if (err == 0) {
-		printf(PHASE_PROMPT "fdt_shrink_to_minimum err - %d\n", err);
-		return;
-	}
+	printf("%s: fdt_shrink_to_minimum is skip\n", __func__);
+	// err = fdt_shrink_to_minimum(fdt_blob, 0);
+	// if (err == 0) {
+	// 	printf(PHASE_PROMPT "fdt_shrink_to_minimum err - %d\n", err);
+	// 	return;
+	// }
 
 	err = arch_fixup_fdt(fdt_blob);
 	if (err) {
-- 
2.34.1

