From eb342a7797e587c2e8a3498ade90ffbb9998ffd7 Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Tue, 9 Jun 2026 15:12:07 +0900
Subject: [PATCH 22/22] Fix machine type

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 tools/libs/light/libxl_dm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/libs/light/libxl_dm.c b/tools/libs/light/libxl_dm.c
index 6fd6fe6015..edced48d02 100644
--- a/tools/libs/light/libxl_dm.c
+++ b/tools/libs/light/libxl_dm.c
@@ -1561,7 +1561,7 @@ static int libxl__build_device_model_args_new(libxl__gc *gc,
     switch (b_info->type) {
     case LIBXL_DOMAIN_TYPE_PVH:
     case LIBXL_DOMAIN_TYPE_PV:
-        flexarray_append(dm_args, "xenpv");
+        flexarray_append(dm_args, "xenpvh");
         for (i = 0; b_info->extra_pv && b_info->extra_pv[i] != NULL; i++)
             flexarray_append(dm_args, b_info->extra_pv[i]);
         break;
@@ -3561,7 +3561,7 @@ void libxl__spawn_qemu_xenpv_backend(libxl__egc *egc,
     flexarray_vappend(dm_args, "-name",
                       GCSPRINTF("domain-%u", domid), NULL);
     flexarray_append(dm_args, "-nographic");
-    flexarray_vappend(dm_args, "-M", "xenpv", NULL);
+    flexarray_vappend(dm_args, "-M", "xenpvh", NULL);
     flexarray_vappend(dm_args, "-monitor", "/dev/null", NULL);
     flexarray_vappend(dm_args, "-serial", "/dev/null", NULL);
     flexarray_vappend(dm_args, "-parallel", "/dev/null", NULL);
-- 
2.34.1

