From 8afec64185c3f920eefe79cc1b22901c4b3a8a66 Mon Sep 17 00:00:00 2001
From: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
Date: Fri, 31 Oct 2025 00:16:02 +0900
Subject: [PATCH] vhost_xen: Fix build error

Signed-off-by: Yuya Hamamachi <yuya.hamamachi.sx@renesas.com>
---
 drivers/vhost/xen.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/vhost/xen.c b/drivers/vhost/xen.c
index 3b2ae7a3f7f6..30ec142718d2 100644
--- a/drivers/vhost/xen.c
+++ b/drivers/vhost/xen.c
@@ -18,6 +18,7 @@
 #include <xen/xenbus.h>
 #include <xen/interface/memory.h>
 #include <asm/xen/hypercall.h>
+#include <linux/vmalloc.h>
 
 #include "vhost.h"
 
-- 
2.34.1

